### PR TITLE
[move source language] Added Functional Test Integration

### PR DESCRIPTION
**Pull request by tnowacki**
## Motivation

- Added Functional Test Integration with a simple test
- Removed static strings from external facing APIs (leaking is now internal)
- Fixed stdlib sanity check from not being run through the bytecode verifier

## Test Plan

cargo test

### CODE REVIEW

File: language/move-lang/src/lib.rs [hunk 1]
```diff
@@ -214,6 +218,11 @@ fn parse_program(
     Ok((files, res))
 }
 
+// TODO replace with some sort of intern table
+fn leak_str(s: &str) -> &'static str {
+    Box::leak(Box::new(s.to_owned()))
+}
```
Comment (sunshowers) [inline language/move-lang/src/lib.rs]: Ahh, I see.

File: language/move-lang/src/test_utils/mod.rs [hunk 2]
```diff
@@ -26,20 +26,16 @@ impl std::error::Error for StringError {
...
-        files.push(s)
-    }
-    files
+    dirfiles
+        .filter(|path| {
+            path.extension()
+                .map(|ext| ext.to_str().map(|s| s == "move").unwrap_or(false))
+                .unwrap_or(false)
+        })
+        .map(|path| path.into_os_string().into_string().unwrap())
```
Comment (sunshowers) [inline language/move-lang/src/test_utils/mod.rs]: Use `filter_map` instead?

Comment (tnowacki) [reply language/move-lang/src/test_utils/mod.rs]: I don't think filter_map would be cleaner here, as the original check is a boolean and does not produce a `Option<Path>`

In other words, I think using filter_map would just wrap the map lambda in an if/else and not really make it more readable.

Comment (tnowacki) [reply language/move-lang/src/test_utils/mod.rs]: But maybe there is some clever way to check the extension while converting it? unsure

Comment (tnowacki) [reply language/move-lang/src/test_utils/mod.rs]: I have found the clever way

**Pull request closed without merging**